### PR TITLE
core: if verbose, add marks for each protocol command

### LINKED ISSUE #6408: core: if verbose, add marks for each protocol command
would be super

### PR DESCRIPTION
**Pull request by Hoten**
image

https://gist.github.com/Hoten/b5a96f138df2649239499d315b99bf41

Fixes #6408

### CODE REVIEW

**Comment by Hoten** [comment]
`Two measures overlap!` warnings, I guess we can just ignore those.

File: lighthouse-core/gather/driver.js [hunk 1]
```diff
@@ -291,11 +291,25 @@ class Driver {
...
+      const status = {
+        msg: `Devtools Protocol Command ${method}`,
+        id: `lh:driver:sendCommand:${method}`,
+      };
+      if (log.isVerbose()) {
+        log.time(status, 'verbose');
+      }
       try {
         const result = await this._connection.sendCommand(method, ...params);
+        if (log.isVerbose()) {
```
Comment (patrickhulce) [inline lighthouse-core/gather/driver.js]: can we stick this in a `finally` instead?

Comment (Hoten) [reply lighthouse-core/gather/driver.js]: It wouldn't be the same functionality - in `finally`, the resolve/reject cb would happen first.

Comment (patrickhulce) [reply lighthouse-core/gather/driver.js]: Oh I'm sensing a learning opportunity for me! :D 

What situations raise an error or impact the timings in any meaningful way? Until today I thought the callbacks would just be run in a microtask afterwards and resolve/reject is basically a super quick bitflip with a bit of state saved

Comment (Hoten) [reply lighthouse-core/gather/driver.js]: Ok, clearly this just registers a value/promise to resolve to, doesn't actually execute it or anything. Slight confusion :)

**Comment by Hoten** [comment]
Leveraging #6425 would be interesting. Would just need to allow for dynamically setting the mark id, given the args. Would be important for distinguishing between each command.

Then all the marking code would be outside the method itself, and could look more like
```js
if (log.isVerbose()) {
    log.autoMark(Driver, {
        sendCommand: {
            msg: (method) => `Devtools Protocol Command ${method}`,
            id: (method) => `lh:driver:sendCommand:${method}`,
        }
    })
}
```

(can iterate on that, but the important part is that all the tracing code is separated from the implementation)

**Comment by Hoten** [comment]
@brendankenny thoughts on above? If it sounds good I will abandon this PR and implement this in #6425 .

**Comment by brendankenny** [comment]
> If it sounds good I will abandon this PR and implement this in #6425 .

Let's close in favor of that.

**Pull request closed without merging**